### PR TITLE
WRP-12241: Upgraded `eslint-plugin-react` to version `^7.32.2`

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop [stanca.pop@lgepartner.com](mailto:stanca.pop@lgepartner.com)

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
`eslint-plugin-react` v7.32.0 has been released on Jan 11, 2023. 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Upgraded `eslint-plugin-react`  to version ^7.32.2 and modified parserOptions in index file.
Renamed deprecated rule `react/jsx-sort-default-props` with `react/js-sort-default-props`

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Also fixed some lint warnings:
- trailing commas and indentation in index.js
- changed deprecated rule name in strict.js

### Links
[//]: # (Related issues, references)
WRP-12241

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -183,12 +187,12 @@ module.exports = {
 			files: ['**/*.ts?(x)'],
 			parser: '@typescript-eslint/parser',
 			parserOptions: {
-				ecmaVersion: 2018,
+				ecmaVersion: 'latest',
```
Comment (daniel-stoian-lgp) [inline index.js:190]: can you share the background information about this change?

Comment (stanca-pop-lgp) [reply index.js]: I forgot to undo this change. Thank you for noticing!

**Review by daniel-stoian-lgp — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ module.exports = {
 		__DEV__: true,
 		process: true,
 		window: true,
-		document: true,
```
Comment (hong6316) [inline index.js:7]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ module.exports = {
...
+		document: true
```
Comment (hong6316) [inline index.js:7]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 4]
```diff
@@ -155,9 +159,9 @@ module.exports = {
 		'react/jsx-boolean-value': ['warn', 'never'],
 		'react/jsx-key': 'off',
 		'react/jsx-no-bind': ['warn', {
-				ignoreRefs: true,
-				allowArrowFunctions: false,
-				allowBind: false
+			ignoreRefs: true,
+			allowArrowFunctions: false,
```
Comment (hong6316) [inline index.js:163]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 5]
```diff
@@ -155,9 +159,9 @@ module.exports = {
...
+			allowBind: false
```
Comment (hong6316) [inline index.js:164]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 6]
```diff
@@ -186,9 +190,9 @@ module.exports = {
 				ecmaVersion: 2018,
 				sourceType: 'module',
 				ecmaFeatures: {
-					jsx: true,
+					jsx: true
```
Comment (hong6316) [inline index.js:193]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 7]
```diff
@@ -186,9 +190,9 @@ module.exports = {
...
 				},
-				warnOnUnsupportedTypeScriptVersion: true,
+				warnOnUnsupportedTypeScriptVersion: true
```
Comment (hong6316) [inline index.js:195]: Please add this change(lint fix) on Additional Considerations.

File: index.js [hunk 8]
```diff
@@ -221,7 +225,7 @@ module.exports = {
 				}],
 				'no-useless-constructor': 'off',
 				'@typescript-eslint/no-useless-constructor': 'warn'
-			},
+			}
```
Comment (hong6316) [inline index.js:228]: Please add this change(lint fix) on Additional Considerations.

File: strict.js [hunk 9]
```diff
@@ -120,7 +120,7 @@ module.exports = {
 		'react/jsx-indent': ['warn', 'tab'],
 		'react/jsx-indent-props': ['warn', 'tab'],
 		'react/jsx-props-no-multi-spaces': 'warn',
-		'react/jsx-sort-default-props': ['warn',{
+		'react/sort-default-props': ['warn', {
```
Comment (hong6316) [inline strict.js:123]: Please add this change(fix deprecation) on Additional Considerations.

**Pull request merged**